cross-chain/bob/contracts/BurnFromMintTokenPoolUpgradeable.sol

```diff
@@ -0,0 +1,168 @@
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ITypeAndVersion} from "./interfaces/ITypeAndVersion.sol";
+
+interface IBurnMintERC20 is IERC20 {
+    /// @notice Mints new tokens for a given address.
+    /// @param account The address to mint the new tokens to.
+    /// @param amount The number of tokens to be minted.
+    /// @dev this function increases the total supply.
+    function mint(address account, uint256 amount) external;
+
+    /// @notice Burns tokens from the sender.
+    /// @param amount The number of tokens to be burned.
+    /// @dev this function decreases the total supply.
+    function burn(uint256 amount) external;
+
+    /// @notice Burns tokens from a given address..
+    /// @param account The address to burn tokens from.
+    /// @param amount The number of tokens to be burned.
+    /// @dev this function decreases the total supply.
+    function burn(address account, uint256 amount) external;
+
+    /// @notice Burns tokens from a given address..
+    /// @param account The address to burn tokens from.
+    /// @param amount The number of tokens to be burned.
+    /// @dev this function decreases the total supply.
+    function burnFrom(address account, uint256 amount) external;
+}
+
+/// @notice Upgradeable BurnFromMintTokenPool that implements CCIP v1.6.0 interface
+/// @dev This is a working implementation for BOB deployment
+contract BurnFromMintTokenPoolUpgradeable is
+    IPoolV1,
+    ITypeAndVersion,
+    Initializable,
+    OwnableUpgradeable
+{
+    /// @notice The token this pool manages
+    IERC20 public s_token;
+
+    /// @notice Mapping of allowed addresses
+    mapping(address => bool) private s_allowList;
+
+    /// @notice The router address
+    address public s_router;
+
+    /// @notice The RMN proxy address
+    address public s_rmnProxy;
+
+    /// @notice The single supported remote chain selector
+    uint64 public s_supportedRemoteChainId;
+
+    /// @notice The version of this contract
+    string public constant override typeAndVersion =
+        "BurnFromMintTokenPoolUpgradeable 1.6.0";
+
+    /// @notice Events
+    event Burned(address indexed sender, uint256 amount);
+    event Minted(
+        address indexed sender,
+        address indexed recipient,
+        uint256 amount
+    );
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice Burns tokens from the pool
+    function lockOrBurn(
+        Pool.LockOrBurnInV1 calldata lockOrBurnIn
+    ) external override returns (Pool.LockOrBurnOutV1 memory) {
+        require(msg.sender == s_router, "Only router");
+        emit Burned(msg.sender, lockOrBurnIn.amount);
+        // Burn tokens
+        IBurnMintERC20(address(s_token)).burnFrom(
```

Comment:
Are we checking minting permission anywhere? Maybe we're adding permissions in deployment scripts somewhere?